### PR TITLE
[Easy] Add transformPostLowering call to checkGraphCompatibility

### LINKED ISSUE #2190: [ONNXIFI] Various fixes for onnxGetBackendCompatibility
* onnxGetBackendCompatibility calls `glow::lower()` to lower graph nodes to what the backends would expect to see but doesn't delete the original nodes so DCE should be run.
* `verify()` should be run on the graphs created by `onnxGetBackendCompatibility` in order to check that they are indeed valid graphs
* `Backend::transformPreLowering()` should be called on the loaded graphs before lowering

### PR DESCRIPTION
**Pull request by jackm321**
*Description*:
Graphs should go through transformPostLowering in checkGraphCompatibility because this is what will happen in initNet

*Testing*:
ninja check

*Documentation*:
Fixes #2190

### CODE REVIEW

**Comment by jackm321** [comment]
I'll need to add DCE to this.

File: lib/Onnxifi/Base.cpp [hunk 1]
```diff
@@ -54,6 +54,10 @@ onnxStatus BackendId::checkGraphCompatibility(const void *onnxModel,
 
   glow::lower(function, /* loweredMap */ nullptr, glowBackend_.get());
 
+  glowBackend_->transformPostLowering(function, CompilationMode::Infer);
```
Comment (jfix71) [inline lib/Onnxifi/Base.cpp]:
```suggestion
  if (glowBackend_->transformPostLowering(function, CompilationMode::Infer)) {
    glow::DCE(function);
  }
```

**Comment by jackm321** [comment]
@jfix71 is this ok now?

File: lib/Onnxifi/Base.cpp [hunk 2]
```diff
@@ -54,6 +54,12 @@ onnxStatus BackendId::checkGraphCompatibility(const void *onnxModel,
 
   glow::lower(function, /* loweredMap */ nullptr, glowBackend_.get());
 
+  // Call the backend's transformPostLowering to match the normal compilation
+  // pipeline then DCE any nodes that are no longer needed
```
Comment (jfix71) [inline lib/Onnxifi/Base.cpp]: nit: period at end of sentence 😃

**Pull request merged**